sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationEditFragment.java

```diff
@@ -0,0 +1,220 @@
+
+		countries = InfrastructureDaoHelper.loadCountries();
+		initialRegions = InfrastructureDaoHelper.loadRegionsByServerCountry();
+		allDistricts = InfrastructureDaoHelper.loadAllDistricts();
+		initialResponsibleDistricts = InfrastructureDaoHelper.loadDistricts(record.getResponsibleRegion());
+		initialResponsibleCommunities = InfrastructureDaoHelper.loadCommunities(record.getResponsibleDistrict());
+		initialDistricts = InfrastructureDaoHelper.loadDistricts(record.getResponsibleRegion());
+		initialCommunities = InfrastructureDaoHelper.loadCommunities(record.getResponsibleDistrict());
+	}
+
+	@Override
+	protected void onLayoutBinding(FragmentImmunizationEditLayoutBinding contentBinding) {
+		contentBinding.setData(record);
+
+		contentBinding.setYesNoUnknownClass(YesNoUnknown.class);
+
+		contentBinding.immunizationCountry.initializeSpinner(countries);
+
+		InfrastructureDaoHelper.initializeRegionFields(
+			contentBinding.immunizationResponsibleRegion,
+			initialRegions,
+			record.getResponsibleRegion(),
+			contentBinding.immunizationResponsibleDistrict,
+			initialResponsibleDistricts,
+			record.getResponsibleDistrict(),
+			contentBinding.immunizationResponsibleCommunity,
+			initialResponsibleCommunities,
+			record.getResponsibleCommunity());
+
+		InfrastructureDaoHelper.initializeRegionFieldListeners(
+			contentBinding.immunizationResponsibleRegion,
+			contentBinding.immunizationResponsibleDistrict,
+			record.getResponsibleDistrict(),
+			contentBinding.immunizationResponsibleCommunity,
+			record.getResponsibleCommunity(),
+			null,
+			null,
+			null,
+			null,
+			null,
+			() -> false);
+	}
+
+	@Override
+	public void onAfterLayoutBinding(final FragmentImmunizationEditLayoutBinding contentBinding) {
+
+		// Initialize ControlSpinnerFields
+		contentBinding.immunizationDisease.initializeSpinner(diseaseList);
+		contentBinding.immunizationImmunizationStatus.initializeSpinner(immunizationStatusList);
+		contentBinding.immunizationImmunizationManagementStatus.initializeSpinner(immunizationManagementStatusList);
+		contentBinding.immunizationMeansOfImmunization.initializeSpinner(meansOfImmunizationList);
+
+		// Initialize ControlDateFields
+		contentBinding.immunizationReportDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationRecoveryDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationStartDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationEndDate.initializeDateField(getFragmentManager());
+		contentBinding.immunizationLastInfectionDate.initializeDateField(getFragmentManager());
+
+		contentBinding.immunizationMeansOfImmunization.addValueChangedListener(e -> {
+			if (e.getValue() == MeansOfImmunization.OTHER || e.getValue() == MeansOfImmunization.RECOVERY) {
+				contentBinding.immunizationImmunizationManagementStatus.setValue(ImmunizationManagementStatus.COMPLETED);
```

Comment:
Vaccination fields and headings have to be hidden when the means of immunization is not VACCINATION or VACCINATION_RECOVERY (see next comment)